src/command/cmd_funcs.c

```diff
@@ -7579,7 +7589,59 @@ cmd_pgp(ProfWin* window, const char* const command, gchar** args)
     }
 
     if (g_strcmp0(args[0], "sendfile") == 0) {
-        _cmd_set_boolean_preference(args[1], command, "Sending unencrypted files using /sendfile while otherwise using PGP", PREF_PGP_SENDFILE);
+        _cmd_set_boolean_preference(args[1], "Sending unencrypted files using /sendfile while otherwise using PGP", PREF_PGP_SENDFILE);
+        return TRUE;
+    }
+
+    if (g_strcmp0(args[0], "sendpub") == 0) {
+        jabber_conn_status_t conn_status = connection_get_status();
+        if (conn_status != JABBER_CONNECTED) {
+            cons_show("You must be connected to share your PGP public key.");
+            return TRUE;
+        }
+
+        if (window->type != WIN_CHAT && args[1] == NULL) {
+            cons_show("You must set recipient in an argument or use this command in a regular chat window to share your PGP key.");
+            return TRUE;
+        }
+
+        ProfChatWin* chatwin = NULL;
+
+        if (args[1]) {
+            char* contact = args[1];
+            char* barejid = roster_barejid_from_name(contact);
+            if (barejid == NULL) {
+                barejid = contact;
+            }
+
+            chatwin = wins_get_chat(barejid);
+            if (!chatwin) {
+                chatwin = chatwin_new(barejid);
+            }
+            ui_focus_win((ProfWin*)chatwin);
+        } else {
+            chatwin = (ProfChatWin*)window;
+            assert(chatwin->memcheck == PROFCHATWIN_MEMCHECK);
+        }
+
+        ProfAccount* account = accounts_get_account(session_get_account_name());
+
+        if (account->pgp_keyid == NULL) {
+            cons_show_error("Please, set the PGP key first using /account set %s pgpkeyid <pgpkeyid>. To list pgp keys, use /pgp keys.", account->name);
+            account_free(account);
+            return TRUE;
+        }
```

Comment:
Would it maybe make more sense to do this check before changing the window focus?

Reply:
window focus?